changelog/fragments/helm-deepequals.yaml

```
@@ -0,0 +1,4 @@
+entries:
+  - description: >
+      For Helm-based operators, fixed a bug where getting a new helm release revision for every reconciliation, even when there are no changes at all.
```

Be specific about the fix, since a previous patch said something similar.
```suggestion
      For Helm-based operators, fixed release equality comparison such that number values are compared and not their types to avoid unnecessary reconciliations.
```